packages/@n8n/db/src/repositories/user.repository.ts

```diff
@@ -181,4 +146,139 @@ export class UserRepository extends Repository<User> {
 			},
 		});
 	}
+
+	private applyUserListSelect(
+		queryBuilder: SelectQueryBuilder<User>,
+		select: Array<keyof User> | undefined,
+	): SelectQueryBuilder<User> {
+		if (select !== undefined) {
+			if (!select.includes('id')) {
+				select.unshift('id'); // Ensure id is always selected
```

Comment:
I'm not sure this is still necessary ? Do we need the id ?

Reply:
Are you sure ? I tested locally and it works without it (and if i set the id as optional in userListItemSchema).
The call I tested: http://localhost:5678/rest/users?select=firstName